agent/task_context.go

```diff
@@ -71,6 +77,47 @@ func (tc *taskContext) addFailingCommand(cmd command.Command) {
 	tc.otherFailingCommands = append(tc.otherFailingCommands, cmd)
 }
 
+func (tc *taskContext) addTaskCommandCleanups(cleanups []internal.CommandCleanup) {
+	tc.Lock()
+	defer tc.Unlock()
+
+	tc.taskCleanups = append(tc.taskCleanups, cleanups...)
+}
+
+func (tc *taskContext) addSetupGroupCommandCleanups(cleanups []internal.CommandCleanup) {
+	tc.Lock()
+	defer tc.Unlock()
+
+	tc.setupGroupCleanups = append(tc.setupGroupCleanups, cleanups...)
+}
+
+// RunCleanupCommands runs all the cleanup commands that have been added to the
+// task config. This should be called right after setup group, teardown group,
```

Comment:
Looks like this comment needs to be updated, since we're allowing setup group to live until the task group ends?